Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.


In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import chi2_contingency

In [2]:
data=pd.read_csv('E:/Data science/Assignments_Data_science/hypothesis testing/BuyerRatio.csv')

In [3]:
data.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

Assume null hypothesis -Sale of the products in all the regions are same for male and female. Ho= independance of the categorical value.
Alternate hypothesis- sale of the products in all the regions will not be same for male and female . Ha = dependance of categorical value.

In [9]:
#create an array
df=np.array([[50,142,131,70],[435,1523,1356,750]])
df

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [11]:
chi2_contingency(df)
#values are [0]=chi2_value, [1]=p_value, [3]=dof

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [12]:
chi2_contingency(df)[1]

0.6603094907091882

As (p_value=0.66)>(chi2_value) we fail to reje